### ISSUE TITLE
Does not handle return from okta after password reset

### ISSUE THREAD

**Issue by dziedrius** [okta/okta-aspnet#170]
## Current behavior
User starts login flow in the app and then continues to okta hosted page, where chooses forgot password option.
After clicking reset password link in email, user is redirected to okta, where he provides new password and after that user is redirected back to the app.
Problem is, that user is redirected using HTTP 302 to:
https://localhost:44314/authorization-code/callback?iss=https://{oktadomain}/oauth2/{id}&type_hint=PASSWORD_RECOVERY&session_hint=AUTHENTICATED&login_hint=user@email.com

and okta middleware does nothing with it, hence 404 page is shown.

## Expected behavior

I guess there might be two approaches:
1. Password reset from SP standpoint is implementation detail, hence SP should not care and should not get related requests to password reset (or activation or whatever).
Also OIDC RFC (https://tools.ietf.org/html/rfc6749#section-4.1.2) states:

> 4.1.2.  Authorization Response
>...
> code - required
> state - required
> ...
> The client MUST ignore unrecognized response parameters.

So best would be remove this behavior from okta at all, as in my opinion it violates OIDC protocol.

2. Fix okta middleware to handle it and document behavior for cases, where for one or another reason other middleware is used (like standard asp.net OIDC middleware, this applies to other platforms too - for example java with springboot security).
I've asked okta's support to help with documentation how it should work, they answer basically was "figure it out yourself".

## Minimal reproduction of the problem with instructions
1. Download okta's asp.net example.
2. Run the example, click on login button
3. Pick forgot password option, enter user email
4. Click link in received email
5. Change the password.
6. Notice, that use is redirected to 404 page.

**Comment by andriizhegurov-okta** [comment]
Hi @dziedrius,
Thank you for reporting the issue! I added an item to our backlog to investigate the issue and take needed actions.
I will comment here as soon as we have new information or questions. 
Thanks.

**Comment by andriizhegurov-okta** [comment]
@dziedrius meanwhile you might want to take a look at this similar issue and check if your App is configured as it's suggested by @bryanapellanes-okta: https://github.com/okta/samples-aspnet/issues/37

**Comment by dziedrius** [comment]
@andriizhegurov-okta - thanks, that link helped me to pinpoint the issue.

Stuff below is let steam off and may be that will help someone in the future:

When creating new app you have this:
![image](https://user-images.githubusercontent.com/772162/112171728-c484f780-8bfc-11eb-85a8-109f925a4a81.png)
After creating app, if you click on edit to see this:
![image](https://user-images.githubusercontent.com/772162/112171961-f72ef000-8bfc-11eb-8809-96d6bd8372f3.png)
It silently copied login redirect url to field, that was not mentioned when creating app - login initiate url.
And that login initiate link is used for redirect after resetting password.
Also notice, that login initiate url has no question mark, probably if I've searched for it, I could find clues, but I did not (as I haven't noticed it).

Interestingly enough, in support case they did not recognized the cause and after couple weeks of ping pong settled on suggestions to follow instructions here: 
https://devforum.okta.com/t/reset-password-flow-doesnt-work-with-oidc-app/8395
which nobody should follow and someone needs to write correct solution.

I think this should be better documented - may be in examples too, may be type_hint and session_hint usage/possible values should be documented too as I've used those to search for solution.
